internal/driver/glfw/loop.go

```diff
@@ -120,8 +120,14 @@ func (d *gLDriver) runGL() {
 					continue
 				}
 
-				if w.shouldExpand {
+				w.viewLock.RLock()
```

Comment:
What if we captured the viewport once at the start of the loop?
```go
w.viewLock.Lock()
v := w.viewport
w.viewLock.Unlock()

if v == nil {
	continue
}
if v.ShouldClose() {
	//...
}
//...
```

Reply:
That means gaining a write lock 60 * windowCount times every second instead of only when windows are closing